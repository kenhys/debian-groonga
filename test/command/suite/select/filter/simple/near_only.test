table_create Logs TABLE_NO_KEY
column_create Logs message COLUMN_SCALAR Text

load --table Logs
[
{"message": "a     b c d e f g h i j k"},
{"message": "a x   b c d e f g h i j k"},
{"message": "a x x b c d e f g h i j k"}
]

table_create Terms TABLE_PAT_KEY ShortText \
  --default_tokenizer TokenBigram \
  --normalizer NormalizerAuto
column_create Terms index COLUMN_INDEX|WITH_POSITION Logs message

log_level --level info
#@add-important-log-levels info
select Logs --filter 'message *N "a k"'
#@remove-important-log-levels info
log_level --level notice
