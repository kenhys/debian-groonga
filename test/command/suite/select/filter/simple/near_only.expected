table_create Logs TABLE_NO_KEY
[[0,0.0,0.0],true]
column_create Logs message COLUMN_SCALAR Text
[[0,0.0,0.0],true]
load --table Logs
[
{"message": "a     b c d e f g h i j k"},
{"message": "a x   b c d e f g h i j k"},
{"message": "a x x b c d e f g h i j k"}
]
[[0,0.0,0.0],3]
table_create Terms TABLE_PAT_KEY ShortText   --default_tokenizer TokenBigram   --normalizer NormalizerAuto
[[0,0.0,0.0],true]
column_create Terms index COLUMN_INDEX|WITH_POSITION Logs message
[[0,0.0,0.0],true]
log_level --level info
[[0,0.0,0.0],true]
select Logs --filter 'message *N "a k"'
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        2
      ],
      [
        [
          "_id",
          "UInt32"
        ],
        [
          "message",
          "Text"
        ]
      ],
      [
        1,
        "a     b c d e f g h i j k"
      ],
      [
        2,
        "a x   b c d e f g h i j k"
      ]
    ]
  ]
]
#|i| [object][search][index][key][near] <Terms.index>
#|i| grn_ii_sel > (a k)
#|i| n=2 (a k)
#|i| exact: 2
#|i| hits=2
log_level --level notice
[[0,0.0,0.0],true]
